x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopDatafeedAction.java

```diff
@@ -50,35 +48,36 @@
         StopDatafeedAction.Response, StopDatafeedAction.Response> {
 
     private final PersistentTasksService persistentTasksService;
+    private final DatafeedConfigProvider datafeedConfigProvider;
 
     @Inject
     public TransportStopDatafeedAction(Settings settings, TransportService transportService, ThreadPool threadPool,
                                        ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver,
-                                       ClusterService clusterService, PersistentTasksService persistentTasksService) {
+                                       ClusterService clusterService, PersistentTasksService persistentTasksService,
+                                       DatafeedConfigProvider datafeedConfigProvider) {
         super(settings, StopDatafeedAction.NAME, threadPool, clusterService, transportService, actionFilters,
                 indexNameExpressionResolver, StopDatafeedAction.Request::new, StopDatafeedAction.Response::new,
                 MachineLearning.UTILITY_THREAD_POOL_NAME);
         this.persistentTasksService = persistentTasksService;
+        this.datafeedConfigProvider = datafeedConfigProvider;
+
     }
 
     /**
      * Resolve the requested datafeeds and add their IDs to one of the list
      * arguments depending on datafeed state.
      *
-     * @param request The stop datafeed request
-     * @param mlMetadata ML Metadata
+     * @param expandedDatafeedIds The expanded set of IDs
      * @param tasks Persistent task meta data
      * @param startedDatafeedIds Started datafeed ids are added to this list
      * @param stoppingDatafeedIds Stopping datafeed ids are added to this list
      */
-    static void resolveDataFeedIds(StopDatafeedAction.Request request, MlMetadata mlMetadata,
+    static void resolveDataFeedIds(Set<String> expandedDatafeedIds,
```

Comment:
As this is no longer expanding ids, we should rename. Or even remove completely?

Reply:
I'm not sure what you mean. The function still needs to call `addDatafeedTaskIdAccordingToState` for each Id I've just changed the function so the set of Ids is a parameter rather than passing the MlMetatdata and reading from that.

Reply:
The `resolve` bit in the name was there to communicate it was expanding ids. It no longer does that. So we should change the method name accordingly. If we keep the method around, then it'd be more a `sortDatafeedIdsBasedOnState` or something.

Reply:
Ok I see what you mean I renamed to `sortDatafeedIdsByTaskState`